### ISSUE TITLE
A few late-game recipe suggestions

### ISSUE THREAD

**Issue by Kataiser9** [Haggle1996/RevolutionPack#529]
[sorry, this is kind of long]

**TLDR: elevator requires space progression, TTKami wand requires eldritch stuff, Angel Ring has an endgame recipe, possibly unlock Green Heart Canisters for very late quest rewards.**

These are just a few things I thought about during the course of playing this pack, which I think could improve the experience. I think these all stay true to the spirit of R3, while encouraging exploration and cross-mod stuff a bit more.

**Enviromine elevator:** This pack clearly has had a lot of focus/effort put into the space aspect - getting into space is a big achievement, but it's also a very expensive undertaking that gets you nothing but cool views. In theory it's a good place to get ReactorCraft ores en masse, but practically speaking it's much faster and easier to take a quick trip to the cave dimension. I propose replacing the bedrock picks in the elevator bottom recipe with a titanium pick and two titanium shards (not ingots, which can be bee-produced). This would make space a real part of endgame progression, which IMO would be pretty rad. (Side effect: Arc Furnace is more valuable, since it takes longer before you can get easy vanilla ores.)

**TTKami wand:** As it stands, it's quite easy to straight-up skip the most interesting and explore-y part of Thaumcraft progression (eldritch stuff and the Outer Lands) by spamming research until you can beeline TTKami's uber-wand, which essentially just require a couple nether stars and a Thaumium/Silverwood wand. Proposed change: in the ichorium wand cap craft, replace the central Ichorium with a Primordial Pearl. (Only catch here is that the 'Nomicon doesn't automatically update when you change recipes. I could probably get around this by rewriting that research entry with Modtweaker.)

**Potential high-end Angel Ring recipe:** (Not sure how this'll be received, but I thought I'd plug it anyway.) Currently the only long-duration flight options in this pack are jetpacks, the Thaumostatic Harness, Draconic Evolution armor, and the very ugly TTKami gear. All are practical, but it seems unfortunate that a lot of cool, customizable high-end armor sets (Thaumium Fortress, Primordial, Terrasteel, Dark Steel) become not-really-worth-using as soon as you get an armored jetpack. Could be cool to have a very late-game Angel Ring recipe for the sake of building convenience and style. The recipe I had in mind uses a Wyvern Core, a HEE Rune of Agility, titanium shards from the Asteroids, and two Uprising foci. (Hell, if you were feeling _really_ mean, you could replace the latter with tritium canisters...)

And finally, since I'm working on the quest book now: at the risk of being a little silly, I had the idea of having ten "uberquests" at the end (visit Barnard's Star, complete the ChC dimension monument, beat Gaia Guardian II, etc) that each reward a green heart canister, which are currently disabled. I'm not sure if those are part of the disabled "gear" module or just disabled individually; the latter would be simple to change, while the former wouldn't be quite so much. Just a goofy detail idea I had, though; I'm not too invested in this one.

Should any of these be accepted, I can whip up the appropriate scripts and submit them as a PR or just paste 'em here.

**Comment by Aivech** [comment]
Green heart canisters don't have a recipe in stock TiCon unless the pack creator adds them artificially.

I do have a few suggestions, though:

The Botania Flugel Tiara already allows creative-like flight without an armor slot. What many people don't seem to know is that having the Eye of the Flugel greatly extends the flight time.

There are other ways to travel long distances quickly that do not involve an armor slot or flight:
- Botania sashes
- ChromatiCraft pendants (speed)
- ChromatiCraft abilities

Using primordial pearls as a balancing item is, in my honest opinion, not a good design choice, for two reasons:
- From experience, the Outer Lands are only fun once.
- There are already multiple items from other mods that require primordial pearls, like the Thaumic Horizons animation focus and the Soul Beacon from the same. Getting a new dungeon to generate requires traveling long distances to find an intact obelisk that won't drop you into the void.

Instead, use something else exclusive to the Outer Lands, like crusted stone (spider rooms) or ancient stone pedestals (knowledge fragment rooms).

**Comment by Kataiser9** [comment]
Whoa, you commented right as I came back to check my email.

Flugel Tiara allows creative-like flight for all of 30 seconds, assuming it hasn't been changed again recently. (Double-checked the Botania changelog to be sure.)

As for long-distance travel, that's true - but that's not really what I'm concerned with; it's more the ease of mobility for building, getting around a large base without placing 50 travel anchors, etc.

What do you think might be the downside of having the Angel Ring accessible and expensive? Honest question - it seems like "being too powerful late-game" wouldn't really be a concern, considering how crazy godlike ChromatiCraft gets.

**Comment by Aivech** [comment]
> Flugel Tiara allows creative-like flight for all of 30 seconds, assuming it hasn't been changed again recently. (Double-checked the Botania changelog to be sure.)

Have you tried using it whilst having the Eye of the Flugel on your person?

**Comment by Kataiser9** [comment]
Nope! I assumed that was just for the teleporty function, I hadn't actually considered that it might do something else. Welp, lesson learned <_<

**Comment by Haggle1996** [comment]
Couple notes:

The Botania trinkets, such as the flugal tiara, won't work if just given as a quest reward -- you have to unlock the achievement first.

As far as Reika's mods go, you cannot reward the player with anything that breaks progression, e.g., give them ChromatiCraft trinkets before they can make it themselves, or give them bedrock ingots before they've made bedrock ingots (but giving them some more as a reward for making a few is alright, as it's been explained to me. Also, Reika request that no spoilers are given in the a quest book for CrC.

**Comment by Haggle1996** [comment]
Also, since I added Draconic Evolution, there's even more OP end game items.

**Comment by Kataiser9** [comment]
(And now I'm unintentionally spamming.)

Haggle, good to know about the Botania trinkets and Reika's mods, although I wouldn't do that stuff anyway - even for non-Reika mods I always avoid rewarding anything beyond what the player can already explicitly make. Glad you brought up the "no spoilers" bit, though, as I wasn't aware of that request.

Honestly, I might just not do a ChC chapter at all, given the heavy "discovery" theme of the mod - let people figure things out through lexicon info, trial and error, and Google/Github like normal. It wouldn't be right to do an overly in-depth/hand-holdy questline, and if the quests are too sparse then they don't serve much purpose at all.

**Comment by Aivech** [comment]
> The Botania trinkets, such as the flugal tiara, won't work if just given as a quest reward -- you have to unlock the achievement first.

That's just the relics you get from the Dice of Fate, actually, which are:
- Ring of Odin - resistance to several different damage types (drowning, fall damage, and fire IIRC), as well as a 10-heart health boost.
- Ring of Thor - Makes the Terra Shatterer considerably more powerful.
- Ring of Loki - Places blocks in a set pattern. Useful if you know how to use it, annoying if you don't.
- Fruit of Grisaia - Eat forever, as long as you have Mana.
- Eye of the Flugel - Teleport to one place. Buffs the Flugel Tiara.
- Key of King's Law - Summons explody glowing weapons out of...somewhere...and throws them at wherever you're looking when you release it.

As well as the Ring of the Aesir, crafted from the three relic rings, which combines the functions of all three.

**Comment by Haggle1996** [comment]
Thank you, @DaMachinator. I remembered the relics didn't work, but confused what was a relic and what wasn't. Memory isn't as sharp as it once was. :P

**Comment by Haggle1996** [comment]
As to your original recipe changes, the only one that concerns me is the change to the elevator. Originally, other planets served as the place to go to get ores, but this caused a few problems.
1. On servers, the Galacticraft planets don't behave well TPS-wise when players start visiting them.
2. Many players want to build a space station or base on the planets -- introducing new ores (onto the moon, for example) frequently meant painful retrogeneration. Using the cave dimension as the mining dimension allows me to add new ores with a simple upgrade system (just delete the cave dim's DIM folder).
3. In the same vein, server operators want an easily destroyed and recreated mining dimension -- space isn't that.

That said, it wouldn't be hard to introduce 'hard-mode' configs, like IE expert, but I'd like it to be more significant than just access to the cave dimension.

**Comment by Kataiser9** [comment]
Interesting. Those are all fair points.
1 - is the TPS issue alleviated with pregeneration, or is it something not related to world gen? Though I do know that pregenning 3 extra dimensions would be kind of a pain.
As for 2 and 3, I was thinking that GC planets wouldn't be long-term mining worlds, just a place to tech up in a bit before you get the elevator. After all, once you get to the moon, you're not that far off from being able to reach the asteroids too.

A different set of "hard-mode" configs could be quite interesting, though I realize it would be no small undertaking. <_<

**Comment by Haggle1996** [comment]
If you create scripts for these, I'll get it included.

**Comment by Kataiser9** [comment]
Right, here you go - I've attached the new scripts; everything looks fine in NEI. Wasn't sure if you wanted to use all three or hold off on some, so I just wrote 'em all.

[new scripts.txt](https://github.com/Haggle1996/RevolutionPack/files/520457/new.scripts.txt)

Also, something I completely forgot to reply to earlier:

> Green heart canisters don't have a recipe in stock TiCon unless the pack creator adds them artificially.

I may have been unclear - I wasn't looking to add a recipe for them, just to add them as a few specific high-end quest rewards. The current problem isn't that there's no recipe, it's that they are (AFAICT) straight-up disabled, and don't show up in NEI. Not sure if there's an easy fix to this or not, as I haven't looked closely at the TiCon config stuff, but it's not a huge deal either way.

**Comment by Haggle1996** [comment]
I probably disabled them; I can look at the configs, see about re-enabling them.

Thank you for the recipes. I'll definitely get the latter two included. I'm still on the fence about the elevator, particularly since Techguns adds ore-dict compatible titanium ingots, which would be confusing if I don't remove one or the other from the dictionary. Let me noodle on that one some more.

**Comment by Kataiser9** [comment]
Sure. Since my elevator recipe uses shards specifically, Techguns titanium would be irrelevant from a "practical" point of view. I can see how it might be a little weird-looking from a user's perspective, that this one specific type of titanium is needed, though I doubt it'd be too much of a problem.

Alternatively, one could replace the titanium shards with silk-touched Ilmenite Ore or something. Kind of a clumsy solution, but it could work.

Let me know when you've decided one way or another so I can plan late-game tech quests accordingly. :D

**Comment by Haggle1996** [comment]
These have been merged in; as Galacticraft is getting removed in the next update (replaced with Advanced Rocketry), I've jiggered the recipes slightly. I'll post the specific change when I've figured out a good progression. Likely it will be tritanium, as this is only obtainable from an Advanced Rocketry mining mission.

**Comment by Haggle1996** [comment]
I elected for dilithium crystals, since you need to travel to space to get them.
